drivers/display/display_rm68200.c

```diff
@@ -0,0 +1,1367 @@
+	}
+}
+
+static int rm68200_blanking_on(const struct device *dev)
+{
+	const struct rm68200_config *config = dev->config;
+
+	if (config->bl_gpio.port != NULL) {
+		return gpio_pin_set_dt(&config->bl_gpio, 0);
+	} else {
+		return -ENOTSUP;
+	}
+}
+
+static int rm68200_set_brightness(const struct device *dev,
+				  const uint8_t brightness)
+{
+	LOG_WRN("Set brightness not implemented");
+	return -ENOTSUP;
+}
+
+static int rm68200_set_contrast(const struct device *dev,
+				const uint8_t contrast)
+{
+	LOG_ERR("Set contrast not implemented");
+	return -ENOTSUP;
+}
+
+static int rm68200_set_pixel_format(const struct device *dev,
+				    const enum display_pixel_format pixel_format)
+{
+	const struct rm68200_config *config = dev->config;
+
+	if (pixel_format == config->pixel_format) {
+		return 0;
+	}
+	LOG_ERR("Pixel format change not implemented");
+	return -ENOTSUP;
+}
+
+static int rm68200_set_orientation(const struct device *dev,
+				   const enum display_orientation orientation)
+{
+	if (orientation == DISPLAY_ORIENTATION_NORMAL) {
+		return 0;
+	}
+	LOG_ERR("Changing display orientation not implemented");
+	return -ENOTSUP;
+}
+
+static void rm68200_get_capabilities(const struct device *dev,
+				     struct display_capabilities *capabilities)
+{
+	const struct rm68200_config *config = dev->config;
+
+	memset(capabilities, 0, sizeof(struct display_capabilities));
+	capabilities->x_resolution = config->panel_width;
+	capabilities->y_resolution = config->panel_height;
+	capabilities->supported_pixel_formats = config->pixel_format;
+	capabilities->current_pixel_format = config->pixel_format;
+	capabilities->current_orientation = DISPLAY_ORIENTATION_NORMAL;
+}
+
+static const struct display_driver_api rm68200_api = {
+	.blanking_on = rm68200_blanking_on,
+	.blanking_off = rm68200_blanking_off,
+	.write = rm68200_write,
+	.read = rm68200_read,
+	.get_framebuffer = rm68200_get_framebuffer,
+	.set_brightness = rm68200_set_brightness,
+	.set_contrast = rm68200_set_contrast,
+	.get_capabilities = rm68200_get_capabilities,
+	.set_pixel_format = rm68200_set_pixel_format,
+	.set_orientation = rm68200_set_orientation,
+};
+
+static int rm68200_init(const struct device *dev)
+{
+	const struct rm68200_config *config = dev->config;
+	uint8_t param[2];
+	int ret;
+	struct mipi_dsi_device mdev;
+
+	if ((config->reset_gpio.port != NULL) && (config->pwr_gpio.port != NULL)) {
+		ret = gpio_pin_configure_dt(&config->reset_gpio, GPIO_OUTPUT_INACTIVE);
+		if (ret < 0) {
+			LOG_ERR("Could not configure reset GPIO (%d)", ret);
+			return ret;
+		}
+		ret = gpio_pin_configure_dt(&config->pwr_gpio, GPIO_OUTPUT_INACTIVE);
+		if (ret < 0) {
+			LOG_ERR("Could not configure power GPIO (%d)", ret);
+			return ret;
+		}
+
+		/* pwr:1 -> reset:0 -> reset:1 */
+		gpio_pin_set_dt(&config->pwr_gpio, 1);
+		k_sleep(K_MSEC(1));
+		gpio_pin_set_dt(&config->reset_gpio, 0);
+		k_sleep(K_MSEC(1));
+		gpio_pin_set_dt(&config->reset_gpio, 1);
+		k_sleep(K_MSEC(5));
```

Comment:
magic sequence here: please document these timings

Reply:
I have added some documentation from the datasheet. These were timings from our SDK example.